backend/src/main/java/com/example/user/UserService.java

```diff
@@ -30,6 +32,27 @@ public boolean userExistsByName(String name) {
         return userRepository.existsByName(name);
     }
 
+    public String setRandomPassword(int length, String email) {
+        if(!userRepository.existsByEmail(email)) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "No user with specified email found");
+        }
+        else {
+            final String chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+            SecureRandom random = new SecureRandom();
+            StringBuilder password = new StringBuilder();
+
+            for (int i = 0; i < length; i++)
+            {
+                int randomIndex = random.nextInt(chars.length());
+                password.append(chars.charAt(randomIndex));
+            }
+
+            userRepository.updatePasswordByEmail(password.toString(), email);
+
+            return password.toString();
```

Comment:
W sumie to zadziała ten kod inna sprawa, że można było użyć czegoś co już istnieje 😄 :
https://www.baeldung.com/java-random-string#apachecommons-bounded

Reply:
czyli zmienić, czy może zostać?